src/components/customers/CustomerCreditNotesList.tsx

```diff
@@ -48,10 +57,12 @@ export const CustomerCreditNotesList = ({
 }: CustomerCreditNotesListProps) => {
   const { translate } = useInternationalization()
   const voidCreditNoteDialogRef = useRef<VoidCreditNoteDialogRef>(null)
-  const { data, loading, error, fetchMore } = useGetCustomerCreditNotesQuery({
-    variables: { customerId, limit: 20 },
-    skip: !customerId,
-  })
+  const [getCreditNotes, { data, loading, error, fetchMore, variables }] =
+    useGetCustomerCreditNotesLazyQuery({
+      variables: { customerId, limit: 20 },
+    })
+  const { debouncedSearch, isSearchLoading } = useDebouncedSearch(getCreditNotes, loading)
+  const isLoading = isSearchLoading || loading
```

Comment:
But... I thought we wanted to debounce the `loading`, if you do that it's not debounced anymore 🤔

Reply:
If the search is loading, then `loading` will never be read (`true || false`)

I kept this for the loading on scroll as we don't want to debounce it. Does it make sense ?

Reply:
Logic updated